tests/sentry/tempest/test_tempest.py

```diff
@@ -0,0 +1,113 @@
+from unittest.mock import patch
+
+from sentry.tempest.tasks import fetch_latest_item_id, poll_tempest_crashes
+from sentry.testutils.cases import TestCase
+from sentry.utils import json
+
+
+class TempestTasksTest(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.project = self.create_project()
+        self.credentials = self.create_tempest_credentials(self.project)
+
+    @patch("sentry.tempest.tasks.fetch_latest_id_from_tempest")
+    def test_fetch_latest_item_id_task(self, mock_fetch):
+        mock_fetch.return_value = "20001"
+        fetch_latest_item_id(self.credentials.id)
+
+        self.credentials.refresh_from_db()
+        assert (
+            self.credentials.latest_fetched_item_id == "20001"
+        )  # Since the ID is stored as a string
+        mock_fetch.assert_called_once()
+
+    @patch("sentry.tempest.tasks.fetch_latest_id_from_tempest")
+    def test_fetch_latest_item_id_error(self, mock_fetch):
+        mock_fetch.return_value = "Invalid credentials"
+
+        fetch_latest_item_id(self.credentials.id)
+
+        self.credentials.refresh_from_db()
+        assert self.credentials.message == "Seems like the provided credentials are invalid"
+        assert self.credentials.latest_fetched_item_id is None
+
+    @patch("sentry.tempest.tasks.fetch_latest_id_from_tempest")
+    def test_fetch_latest_item_id_ip_not_allowlisted(self, mock_fetch):
+        mock_fetch.return_value = "IP address not allow-listed"
+
+        fetch_latest_item_id(self.credentials.id)
+
+        self.credentials.refresh_from_db()
+        assert self.credentials.message == "Seems like our IP is not allow-listed"
+        assert self.credentials.latest_fetched_item_id is None
+
+    @patch("sentry.tempest.tasks.fetch_latest_id_from_tempest")
+    def test_fetch_latest_item_id_unexpected_response(self, mock_fetch):
+        mock_fetch.return_value = "Some other error"
+
+        fetch_latest_item_id(self.credentials.id)
+
+        self.credentials.refresh_from_db()
+        assert self.credentials.latest_fetched_item_id is None
+        assert self.credentials.message == ""  # No specific message set for unexpected responses
+        mock_fetch.assert_called_once()
+
+    @patch("sentry.tempest.tasks.fetch_latest_id_from_tempest")
+    def test_fetch_latest_item_id_connection_error(self, mock_fetch):
+        mock_fetch.side_effect = Exception("Connection error")
+
+        with self.assertLogs("sentry.tempest.tasks", level="INFO") as cm:
+            fetch_latest_item_id(self.credentials.id)
+
+        self.credentials.refresh_from_db()
+        assert self.credentials.latest_fetched_item_id is None
+        assert self.credentials.message == ""
+        mock_fetch.assert_called_once()
+        assert "Fetching the latest item id failed." in cm.output[0]
+
+    @patch("sentry.tempest.tasks.fetch_crashes_from_tempest")
```

Comment:
```suggestion
    @patch("sentry.tempest.tasks.fetch_crashes_from_tempest.delay")
```

We need to patch the call to `.delay` method, since that is what really gets called